src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -73,23 +78,23 @@ internal Task InitializeChannelsAsync()
             {
                 return Task.CompletedTask;
             }
-            if (_environment.IsLinuxContainerEnvironment())
-            {
-                return Task.CompletedTask;
-            }
             if (string.IsNullOrEmpty(workerRuntime) && _environment.IsPlaceholderModeEnabled())
             {
-                // Only warm up language workers in placeholder mode in worker runtime is not set
-                return Task.WhenAll(_languages.Select(runtime => _languageWorkerChannelManager.InitializeChannelAsync(runtime)));
+                // Only warm up language workers in placeholder mode if worker runtime is not set
+                if (_environment.IsLinuxContainerEnvironment())
+                {
+                    return Task.WhenAll(_linuxLanguages.Select(runtime => _languageWorkerChannelManager.InitializeChannelAsync(runtime)));
+                }
+                return Task.WhenAll(_windowsLanguages.Select(runtime => _languageWorkerChannelManager.InitializeChannelAsync(runtime)));
             }
-            if (_languages.Contains(workerRuntime))
+            if (_windowsLanguages.Contains(workerRuntime))
             {
```

Comment:
same here. When workerruntime is set, if selected language is whitelisted here, language worker should start at the app level.